glambie/data/timeseries.py

```diff
@@ -0,0 +1,118 @@
+
+from dataclasses import dataclass
+from decimal import Decimal
+from decimal import getcontext
+
+from glambie.const.data_groups import GlambieDataGroup
+from glambie.const.regions import RGIRegion
+import numpy as np
+import pandas as pd
+
+
+@dataclass
+class TimeseriesData():
+    """Class to wrap the actual data contents of a Timeseries"""
+    dates: np.ndarray
+    area: np.ndarray
+    changes: np.ndarray
+    errors: np.ndarray
+
+    @property
+    def min_time(self) -> float:
+        return np.min(self.dates)
+
+    @property
+    def max_time(self) -> float:
+        return np.max(self.dates)
+
+    @property
+    def min_change_value(self) -> float:
+        finite_list = np.isfinite(self.changes)
+        return np.min(self.changes[finite_list])
+
+    @property
+    def max_change_value(self) -> float:
+        finite_list = np.isfinite(self.changes)
+        return np.max(self.changes[finite_list])
+
+    @property
+    def temporal_resolution(self) -> float:
+        getcontext().prec = 3  # deal with floating point issues
+        resolution = (Decimal(self.max_time) - Decimal(self.min_time)) / len(self)
+        return float(resolution)
+
+    def __len__(self) -> int:
+        return len(self.dates)
+
+    def as_dataframe(self):
+        return pd.DataFrame({'dates': self.dates,
+                             'changes': self.changes,
+                             'errors': self.errors,
+                             'area': self.area
+                             })
+
+
+class Timeseries():
+    """Class containing a data series and corresponding metadata"""
+    is_data_loaded = False
+
+    def __init__(self, region: RGIRegion = None, data_group: GlambieDataGroup = None, data_filepath: str = None,
```

Comment:
A comment in the constructor would be good about the lazy data loading or presence of data as input